db/indexes_test.go

```diff
@@ -69,6 +71,10 @@ func TestInitializeIndexes(t *testing.T) {
 
 			validateErr := validateAllIndexesOnline(b)
 			require.NoError(t, validateErr, "Error validating indexes online")
+
+			// Drop again
+			dropErr = base.DropAllIndexes(base.TestCtx(t), n1qlStore)
+			require.NoError(t, dropErr, "Error dropping all indexes")
```

Comment:
I think we should only do this if `test.collections == true`

The bucket indexes are expected to be kept around when pooled for the next test to use - they're not recreated in the bucket reader - only emptied

Reply:
The reasons that this doesn't work is that for server 6.x a keyspace is only a bucket, and for 7.x the keyspace needs to be fully qualified if you are not using the `ON` keyword.

We could create a function that gets a fully qualified keyspace:

if 6.x -> `default:bucket`
if 7.x -> `default:bucket._default._default` for unnamed collection
if 7.x -> `default:bucket.scope.collection` for named collection

Reply:
As per https://docs.couchbase.com/server/current/n1ql/n1ql-language-reference/dropindex.html#keyspace-full-index, changing our `EscapedKeyspace` function to match this which will always be correct (if verbose). This seems to pass tests against 6.x (out of jenkins) and 7.x in jenkins.